SICP_book/chapter_1_1.metta

```diff
@@ -0,0 +1,191 @@
+(= (sqr $x) (* $x $x))
+
+!(assertEqual
+    (sqr 21)
+    441)
+
+!(assertEqual
+    (sqr (+ 2 5))
+    49)
+
+!(assertEqual
+    (sqr (sqr 3))
+    81)
+
+(= (sum-of-squares $x $y) (+ (sqr $x) (sqr $y)))
+
+!(assertEqual
+    (sum-of-squares 3 4)
+    25)
+
+(= (f $a) (sum-of-squares (+ $a 1) (* $a 2)))
+
+!(assertEqual
+    (f 5)
+    136)
+
+(= (abs $x)
+    (if (> $x 0)
+        $x
+        (if (== $x 0)
+            0
+            (* $x -1))))
+
+!(assertEqual
+    (abs -5)
+    5)
+
+(= (_abs $x)
+    (if (< $x 0)
+        (* $x -1)
+        $x))
+
+!(assertEqual
+    (_abs -5)
+    5)
+
+(= (>= $x $y) (or (> $x $y) (== $x $y)))
+
+!(assertEqual
+    (>= 6 5)
+    True)
+
+; excercise 1.3
+; Define procedure that takes three numbers as arguments and returns
+; the sum of the squares of the two larger numbers
+
+;I haven't got an idea of function's name
+(= (exercise_1_3 $x1 $x2 $x3)
+  (if (and (>= $x1 $x2) (>= $x3 $x2))
+    (sum-of-squares $x1 $x3)
+    (if (and (>= $x3 $x1) (>= $x2 $x1))
+        (sum-of-squares $x2 $x3)
+        (sum-of-squares $x1 $x2))))
+
+;check function using all permutations of 1 2 and 5
+!(assertEqual
+    (exercise_1_3 1 2 5)
+    29)
+!(assertEqual
+    (exercise_1_3 1 5 2)
+    29)
+!(assertEqual
+    (exercise_1_3 2 1 5)
+    29)
+!(assertEqual
+    (exercise_1_3 2 5 1)
+    29)
+!(assertEqual
+    (exercise_1_3 5 1 2)
+    29)
+!(assertEqual
+    (exercise_1_3 5 2 1)
+    29)
+; -----------------------End of Excercise 1.3----------------------------
+
+(= (a-plus-abs-b $a $b)
+    ((if (> $b 0) + -) $a $b))
+
+!(assertEqual
+    (a-plus-abs-b 5 -5)
+    10)
+
+
+;test to determine whether the interpreter he is faced with is using
+;applicative-order evaluation or normal-order evaluation
+(= (p) (p))
+(= (test $x $y)
+    (if (= $x 0)
+    0
+    $y))
+
+;in case of metta, this test crashes with segmentation fault after a while
+;so it seems that metta is using applicative order evaluation
+;!(test 0 (p))
```

Comment:
Test will not fail if you assign a proper type to `test` for example `(: test (-> Number Atom Atom))`. Thus MeTTa supports both. This support is restricted because only meta-types can enable this. But I would add this type and add a comment that default type of the `test` which is `%Undefined%` will cause using applicative evaluation order and proper type will cause using normal evaluation order.